### PR TITLE
Fix add_interface_address

### PR DESCRIPTION
**Pull request by drossy**
Don't attempt to add_interface_address if there is no dns_name entry in netbox for the ip address

### CODE REVIEW

**Comment by jrha** [comment]
Why? We rely on this working.

**Comment by drossy** [comment]
We don't have our management address in DNS, so there isn't a dns_name in netbox for those ip addresses for us

File: aquilon/netbox2aquilon [hunk 1]
```diff
@@ -306,17 +306,21 @@ def _netbox_copy_interfaces(device):
         addresses = _get_addresses_from_interface(interface)
         for address in addresses:
             # Don't add the primary IP as add_host will do this
-            if address.address != device.primary_ip.address:
+            # If there is no dns_name in netbox for the address do not do this as we have
+            # not done the add_host so there is no primary name for the host, so the fqdn
+            # will be needed by add_interface_address
```
Comment (jrha) [inline aquilon/netbox2aquilon:311]: Why not just move the `add_interface_address` commands to after `add_host`?

File: aquilon/netbox2aquilon [hunk 2]
```diff
@@ -306,17 +306,21 @@ def _netbox_copy_interfaces(device):
...
-                    f'--interface {interface.name}',
-                    f'--fqdn {address.dns_name}' if address.dns_name else '\b',
-                    f'--network_environment {address.vrf}' if address.vrf else '\b',
-                    f'--ip {address.address}',
-                ]))
+                cmd = ['add_interface_address',
+                       f'--machine {device.aq_machine_name}',
+                       f'--interface {interface.name}',
+                       f'--ip {address.address}',
+                       f'--fqdn {address.dns_name}'
```
Comment (jrha) [inline aquilon/netbox2aquilon:319]: We need to be able to support interface addresses that do not have `fqdn`.

**Comment by jrha** [comment]
Ditto, that's why fqdn is optional.

File: aquilon/netbox2aquilon [hunk 3]
```diff
@@ -302,21 +302,27 @@ def _netbox_copy_interfaces(device):
...
-                    f'--machine {device.aq_machine_name}',
-                    f'--interface {interface.name}',
-                    f'--fqdn {address.dns_name}' if address.dns_name else '\b',
-                    f'--network_environment {address.vrf}' if address.vrf else '\b',
-                    f'--ip {address.address}',
-                ]))
+                cmd = ['add_interface_address',
+                       f'--machine {device.aq_machine_name}',
+                       f'--interface {interface.name}',
+                       f'--ip {address.address}']
```
Comment (jrha) [inline aquilon/netbox2aquilon:320]: Can you reformat this as follows? It'll keep the diff a bit cleaner.
```suggestion
                cmd = [
                    'add_interface_address',
                    f'--machine {device.aq_machine_name}',
                    f'--interface {interface.name}',
                    f'--ip {address.address}',
                ]
```

**Review by jrha — APPROVED** [review]
LGTM
Minor request around formatting.

**Comment by drossy** [comment]
Now with improved formatting

**Review by jrha — APPROVED** [review]

**Pull request merged**